WordPress/Classes/ViewRelated/Tools/SettingsTextViewController.h

```diff
@@ -1,13 +1,38 @@
 #import <UIKit/UIKit.h>
 
+
+// Typedef's
+typedef NS_ENUM(NSInteger, SettingsTextModes) {
+    SettingsTextModesText,
+    SettingsTextModesEmail,
+    SettingsTextModesPassword
+};
+
+typedef void (^SettingsTextChanged)(NSString *);
+
+/// Reusable component that renders a UITextField + Hint onscreen. Useful for Text / Password / Email data entry.
+///
 @interface SettingsTextViewController : UITableViewController
 
-@property (nonatomic, copy) void(^onValueChanged)(NSString *);
-@property (nonatomic, copy) void(^onCancel)();
+/// Block to be executed on dismiss, if the value was effectively updated.
+///
+@property (nonatomic, copy) SettingsTextChanged onValueChanged;
+
+/// Sets the Text Input Mode:
+///
+/// - SettingsTextModesText: Default mode
+/// - SettingsTextModesEmail: Will perform Email validation before hitting the callback.
+/// - SettingsTextModesPassword: Secure Text Entry is enabled.
+///
+@property (nonatomic, assign) SettingsTextModes mode;
```

Comment:
I'd like to defer that, until there's actual need to do so (ie. another issue perhaps?).

If that validation method grows, we can either add another object, as you pointed out, or do something with a smaller footprint. Something such as..

```
let validatorsMap [SettingsTextModes : (String) -> (Bool)]
```

(Swift is *way* more expressive!). If that has to be extended to multiple validators, per mode, we can simply have an array of closures.

We should swift the whole project, BTW!